### PR TITLE
just trolling the repo [LibDP Compat]

### PR DESCRIPTION
**Pull request by Yoghurt4C**
re-added libdp compat, should just work (I've accidentally some imports though)

### CODE REVIEW

File: src/main/java/techreborn/compat/libdp/TRTweaker.java [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+import techreborn.init.ModRecipes;
+import techreborn.init.TRContent;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+
+public class TRTweaker implements Driver {
```
Comment (LemmaEOF) [inline src/main/java/techreborn/compat/libdp/TRTweaker.java]: Would better to rename to LibDP's new name scheme (`TRTweaker` -> `TRDriver`) but not mandatory

**Review by LemmaEOF — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -88,6 +91,7 @@ dependencies {
 
 	optionalDependency ("me.shedaniel:RoughlyEnoughItems:5.2.3")
 	disabledOptionalDependency ('com.github.emilyploszaj:trinkets:2.6.0')
+    optionalDependency ("io.github.cottonmc:LibDP:1.1.0+1.16.2")
```
Comment (Sturmlilie) [inline build.gradle]: spurious spacing?

File: src/main/java/techreborn/compat/libdp/TRRecipeParser.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+						}
+					}
+					Identifier id = new Identifier(in);
+					Item item = Registry.ITEM.get(id);
+					if (item == Items.AIR) throw new DPSyntaxError("Failed to get item for input: " + in);
+					stack = new ItemStack(item);
+				}
+				return new StackIngredient(Collections.singletonList(stack), amount, tag, requireEmpty);
+			}
+		} else
```
Comment (Sturmlilie) [inline src/main/java/techreborn/compat/libdp/TRRecipeParser.java]: no braces around final else?

File: src/main/java/techreborn/compat/libdp/TRTweaker.java [hunk 4]
```diff
@@ -0,0 +1,634 @@
...
+	 * @param fluid   The fluid required.
+	 * @param holders The fluid-holding items the fluid can be in, or [] for any.
+	 * @param amount  The amount of fluid needed, or -1 for any.
+	 * @return A prepared fluid ingredient.
+	 */
+	public FluidIngredient createFluidIngredient(String fluid, String[] holders, int amount) {
+		Fluid parsedFluid = DriverUtils.INSTANCE.getRawFluid(fluid);
+		Optional<List<Item>> parsedHolders;
+		Optional<Integer> count;
+		if (holders.length == 0) parsedHolders = Optional.empty();
```
Comment (Sturmlilie) [inline src/main/java/techreborn/compat/libdp/TRTweaker.java]: same mixed braced / braceless style

**Review by Sturmlilie — APPROVED** [review]
lgtm

**Comment by jusvit** [comment]
Not applicable for 1.17 anymore and LibCD is effectively abandoned. 

If you disagree, contest me.

**Pull request closed without merging**